### ISSUE TITLE
Case details do not show up on create/update plan page for Case Triggered plans RVL-1164

### ISSUE THREAD

**Issue by Naima-Bashir** [onaio/reveal-frontend#1278]
The Case Details section is missing from /plans/update/ page for case triggered plans 
![image](https://user-images.githubusercontent.com/61143692/93193008-02392d00-f74f-11ea-9319-195938c0710a.png)

**Comment by p-netm** [comment]
@Naima-Bashir @samkanga do you have an example plan whose case details panel is not showing up.
note we only show that panel for `Case Triggered` plans. @Naima-Bashir :pray: please also add the environment(i.e `stage`/`Thailand-preview`, ... ) to the description or/and a url to a page where we can reproduce the problem.

**Comment by p-netm** [comment]
this one shows the caseDetails section: https://web.reveal-th-preview.smartregister.org/plans/update/b576bbbd-68dc-41f4-9208-8bbcd9c1b3aa

**Comment by samkanga** [comment]
@p-netm its mainly the dynamic plans. I have seen your comment on this being for case triggered plans only. There is however some cases of case triggered plans e.g. https://web.reveal-th-preview.smartregister.org/plans/update/d3335acb-7e42-42e7-abeb-30110f7f50f3

**Comment by p-netm** [comment]
@samkanga @Naima-Bashir [slack conversation thread](https://onaio.slack.com/archives/C01ASKAEZKL/p1600265305020900)
basically we have invalid plans that are missing a crucial field that we need to show the missing section in the issue.

**Comment by moshthepitt** [comment]
Looks like if we take these "invalid" plans into account then the solution for RVL-1164 is to do this:

1. Explain why these are invalid plans to concerned parties
2. Notify concerned parties that invalid plans will be filtered out of the web UI
3. Implement web UI filter for these

**Comment by craigappl** [comment]
Shifting over my findings from slack:

1. There are 23 plans in reveal-th-preview that don't have an fiReason useContext.
2. All of these seem to be modified in the web UI.
3. I checked the Nifi flows and plan creation creates an fiReason for case triggered plans.
4. The FI Reason dropdown in the plan form is not a required field.
5. When a user edits a plan with an fiReason of "Case Triggered" it changes to "Routine" in the web UI before the update is posted to the server.

We need to do the following work.

- [ ] Update the plan templates to make the FI reason a required field.
- [ ] Figure out why the fiReason gets changed when we edit a plan and update that.
- [ ] Update the 23 plans so they now have an fiReason in the server

**Comment by craigappl** [comment]
More comments on the fiReason before and after on 5.

Steps:
1. Find a plan on Reveal thailand preview that has an fiReason such as plan Id https://web.reveal-th-preview.smartregister.org/plans/update/654233df-cd53-4be2-b0fb-b29b6740a65e
Here's the plan json downloaded from the browser (goals and actions removed for brevity. Notice there's an useContext fiReason of "Case Triggered"
```
[
  {
    "identifier": "654233df-cd53-4be2-b0fb-b29b6740a65e",
    "version": "1",
    "name": "A2-Thailand test site BVBD 2-FNCA1-A2-Bx_LNCA1-A2-Bx-2020-09-15-Site",
    "title": "A2 - Thailand test site BVBD 2 - FNCA1-A2-Bx LNCA1-A2-Bx - 2020-09-15 - Site",
    "status": "draft",
    "date": "2020-09-15",
    "effectivePeriod": {
      "start": "2020-09-15",
      "end": "2020-10-05"
    },
    "useContext": [
      {
        "code": "interventionType",
        "valueCodableConcept": "FI"
      },
      {
        "code": "fiStatus",
        "valueCodableConcept": "A2"
      },
      {
        "code": "fiReason",
        "valueCodableConcept": "Case Triggered"
      },
      {
        "code": "opensrpEventId",
        "valueCodableConcept": "c382d8f9-c412-4bce-a790-158c2b930e8a"
      },
      {
        "code": "caseNum",
        "valueCodableConcept": "1000020"
      },
      {
        "code": "taskGenerationStatus",
        "valueCodableConcept": "False"
      }
    ],
    "jurisdiction": [
      {
        "code": "a7baf57d-ad31-46d2-8d28-0c81dd306b09"
      }
    ],
    "serverVersion": 1600189160927,
    "experimental": false
  }
]
```
2. In the developer tools, set the connection to "Offline" so we don't actually post this plan.
3. Change the status to "retired" and save the plan (it fails because you're offline.
4. In the developer tools, see what would have posted to the server in this situation (below, again removed actions and goals)
```
{
  "date": "2020-09-14",
  "effectivePeriod": {
    "end": "2020-10-04",
    "start": "2020-09-14"
  },
  "experimental": false,
  "identifier": "654233df-cd53-4be2-b0fb-b29b6740a65e",
  "jurisdiction": [
    {
      "code": "a7baf57d-ad31-46d2-8d28-0c81dd306b09"
    }
  ],
  "name": "A2-a7baf57d-ad31-46d2-8d28-0c81dd306b09-2020-09-14",
  "status": "retired",
  "title": "A2 - Thailand test site BVBD 2 - FNCA1-A2-Bx LNCA1-A2-Bx - 2020-09-15 - Site",
  "useContext": [
    {
      "code": "interventionType",
      "valueCodableConcept": "FI"
    },
    {
      "code": "fiStatus",
      "valueCodableConcept": "A2"
    },
    {
      "code": "fiReason",
      "valueCodableConcept": "Routine"
    },
    {
      "code": "caseNum",
      "valueCodableConcept": "1000020"
    },
    {
      "code": "opensrpEventId",
      "valueCodableConcept": "c382d8f9-c412-4bce-a790-158c2b930e8a"
    },
    {
      "code": "taskGenerationStatus",
      "valueCodableConcept": "False"
    }
  ],
  "version": 2
}
```
See that the fiReason changed to Routine instead of case triggered

Now, find a plan that doesn't have an fiReason such as https://web.reveal-th-preview.smartregister.org/plans/update/9f882f0c-0a81-4799-8845-86faa967d4b9
Below is the plan information:
```
[
  {
    "identifier": "9f882f0c-0a81-4799-8845-86faa967d4b9",
    "version": "2",
    "name": "A1-a7baf57d-ad31-46d2-8d28-0c81dd306b09-2020-09-04",
    "title": "A1 - Thailand test site BVBD 2 - PatientFirstname20200904 Patientlastname20200904 - 2020-09-04 - Site",
    "status": "active",
    "date": "2020-09-04",
    "effectivePeriod": {
      "start": "2020-09-04",
      "end": "2020-09-24"
    },
    "useContext": [
      {
        "code": "interventionType",
        "valueCodableConcept": "FI"
      },
      {
        "code": "fiStatus",
        "valueCodableConcept": "A1"
      },
      {
        "code": "caseNum",
        "valueCodableConcept": "1000001"
      },
      {
        "code": "opensrpEventId",
        "valueCodableConcept": "553d5cd6-0af1-4e96-bb22-69025b2a8a0f"
      },
      {
        "code": "taskGenerationStatus",
        "valueCodableConcept": "True"
      }
    ],
    "jurisdiction": [
      {
        "code": "a7baf57d-ad31-46d2-8d28-0c81dd306b09"
      }
    ],
    "serverVersion": 1599222893773,
    "experimental": false
  }
]
```
Again, take the browser offline in the developer tools, change the status and click save. See what posts:
```
{

  "date": "2020-09-03",
  "effectivePeriod": {
    "end": "2020-09-23",
    "start": "2020-09-03"
  },
  "experimental": false,
  "identifier": "9f882f0c-0a81-4799-8845-86faa967d4b9",
  "jurisdiction": [
    {
      "code": "a7baf57d-ad31-46d2-8d28-0c81dd306b09"
    }
  ],
  "name": "A1-a7baf57d-ad31-46d2-8d28-0c81dd306b09-2020-09-03",
  "status": "retired",
  "title": "A1 - Thailand test site BVBD 2 - PatientFirstname20200904 Patientlastname20200904 - 2020-09-04 - Site",
  "useContext": [
    {
      "code": "interventionType",
      "valueCodableConcept": "FI"
    },
    {
      "code": "fiStatus",
      "valueCodableConcept": "A1"
    },
    {
      "code": "fiReason",
      "valueCodableConcept": "Routine"
    },
    {
      "code": "caseNum",
      "valueCodableConcept": "1000001"
    },
    {
      "code": "opensrpEventId",
      "valueCodableConcept": "553d5cd6-0af1-4e96-bb22-69025b2a8a0f"
    },
    {
      "code": "taskGenerationStatus",
      "valueCodableConcept": "True"
    }
  ],
  "version": 3
}
```
See that an fiReason of Routine is added to this plan

**Comment by craigappl** [comment]
I updated all of the plans on Thailand Preview to add an fiReason so we no longer get that error.

**Comment by p-netm** [comment]
closed in #1290
